modules/ROOT/pages/database/database-connector-examples.adoc

```diff
@@ -0,0 +1,737 @@
+
+[literal,source,xml,linenums]
+....
+<db:update config-ref="dbConfig">
+    <db:sql><![CDATA[#["UPDATE electronic SET discount = :discount WHERE price > :price"]]]></db:sql>
+	<db:input-parameters>#[{
+        discount: 10,
+        price: 100
+    }]</db:input-parameters>
+</db:update>
+....
+
+The `<![CDATA[...]]>` wrapper allows you to use special characters, such as `>` or `"`, in the query.
+Otherwise, you must use XML-escaped versions of those characters, such as `+&gt;+` and  `+&quot;+`.
+
+When you use the Anypoint Studio visual user interface for connectors instead of the XML code view, you can type special
+characters directly into the `SQL Query Text` box, and Anypoint Studio automatically
+changes the character to its escaped version in the XML view.
+
+== Delete Example
+
+Given the table defined in the <<_insert_example>>, the following operation
+ deletes the record with _id_ 1 from the *electronic* table:
+
+[source,xml,linenums]
+----
+<db:delete config-ref="dbConfig">
+    <db:sql>DELETE FROM electronic WHERE id = :id</db:sql>
+    <db:input-parameters>#[{
+        id: 1
+    }]</db:input-parameters>
+</db:delete>
+----
+
+= Example: Execute Stored Procedures - Mule 4
+
+Invoke stored procedures that combine, input, output, and input-output parameters.
+
+For example:
+
+[source,xml,linenums]
+----
+<!-- Invoke a procedure with input parameters -->
+<db:stored-procedure config-ref="dbConfig">
+    <db:sql>{ call updatePlanetDescription('Venus', :description) }</db:sql>
+    <db:input-parameters>
+        #[{'description' : payload}]
+    </db:input-parameters>
+</db:stored-procedure>
+
+<!-- Invoke a procedure with input-output parameters -->
+<db:stored-procedure config-ref="dbConfig">
+  <db:sql>{ call doubleMyInt(:myInt) }</db:sql>
+  <db:in-out-parameters>
+      <db:in-out-parameter key="myInt" value="3"/>
+  </db:in-out-parameters>
+</db:stored-procedure>
+
+<!-- Invoke a procedure with both input AND output parameters -->
+<db:stored-procedure config-ref="dbConfig">
+    <db:sql>{ call multiplyInts(:int1, :int2, :result1, :int3, :result2) }</db:sql>
+    <db:input-parameters>
+        #[{
+            'int1' : 3,
+            'int2' : 4,
+            'int3' : 5
+        }]
+    </db:input-parameters>
+    <db:output-parameters>
+        <db:output-parameter key="result1" type="INTEGER"/>
+        <db:output-parameter key="result2" type="INTEGER"/>
+        <db:output-parameter key="myInt" type="INTEGER"/>
+    </db:output-parameters>
+</db:stored-procedure>
+----
+
+Many combinations are possible.
+
+[TIP]
+After execution, the resulting values of the output and input-output parameters are available in the operation's result. Leverage DataSense to help you get them.
+
+[NOTE]
+For connector version 1.4.0 or later, make sure that no whitespace is between the name of the stored procedure and the first parentheses.
+
+= Example: Execute DDL Statements - Mule 4
```

Comment:
You list some of these as Mule 4. Is there a reason for that?

Reply:
Not sure. I would assume it is all Mule 4?